invenio_oauthclient/contrib/cern.py

```diff
@@ -209,16 +211,68 @@ def account_setup(remote, token):
 
     response = remote.get(REMOTE_APP_RESOURCE_API_URL)
     user = token.remote_account.user
+    if not user:
+        raise Exception('`user` should not be None, something bad happened!')
 
     if response.status == requests.codes.ok:
         res = get_dict_from_response(response)
-        current_user.info['group'] = fetch_groups(res['Group'])
+
+        # update session object
+        groups_old = current_user.info['group']
+        groups_new = fetch_groups(res['Group'])
+        current_user.info['group'] = groups_new
         current_user.modified = True
         current_user.save()
 
-        if user and not any([user.family_name, user.given_names]):
+        # update DB object
+        modified = False
+
+        # try to add/remove groups
+        try:
```

Comment:
It is there to decide if we do a `current_user.reload()`. Ideally, our `invenio.ext.login.legacy_user` module should know do this on its own.